src/pydrex/deformation_mechanism.py

```diff
@@ -0,0 +1,17 @@
+"""PyDRex: Deformation mechanism enums.
+
+Public symbols:
+- `Regime`
+
+"""
+__all__ = ["Regime"]
```

Comment:
This is not strictly necessary, the only effects are:
- `from foo import *` only imports names from `__all__` [1]
- some documentation and linting tools can rely on this

For example pylint gives an error if something from `__all__` is missing.
I was going to trial this as a way to distinguish public vs internal interface.
Most python standard library modules use this.
However, it does require typing things twice which is a bit annoying.
Let me know if we should rather remove it.

This could be made use of in `__init__.py` files when creating a package, but there are other options.

Reply:
I do not think the code benefits from it, and considering it is annoying to do in the first place, probably best to remove it if you agree; `from foo import *` is not good practice anyway. We will see if it is useful when we deal with `__init__.py` files.

Reply:
OK, I'll also get rid of the `Public symbols` noise, I think I included it for documentation but `help(module)` already prints a list of functions:

```
FUNCTIONS
    Voigt_matrix_to_vector(matrix)
        Create the 21-component Voigt vector equivalent to the 6x6 Voigt matrix.

    Voigt_to_elastic_tensor(matrix)
        Create 4-th order elastic tensor from an equivalent Voigt matrix.

    elastic_tensor_to_Voigt(tensor)
        Create a 6x6 Voigt matrix from an equivalent 4-th order elastic tensor.

    rotate(tensor, rotation)
        Rotate 4-th order tensor using a 3x3 rotation matrix.

    upper_tri_to_symmetric(arr)
        Create symmetric array using upper triangle of input array.

```